lib/dal/src/attribute/value.rs

```diff
@@ -2125,4 +2128,108 @@ impl AttributeValue {
+                            .get_node_weight(order_of_thing_node)
+                            .await?
+                            .get_ordering_node_weight()?;
+
+                        let order = order_node_weight.order();
+
+                        // Find the index of the item in the array to add to path
+                        let index = order
+                            .iter()
+                            .position(|&id| id == attribute_value_id.into())
+                            .ok_or(AttributeValueError::ParentAttributeValueMissing(
+                                attribute_value_id,
+                            ))?;
+                        let name = format!("{0}[{1}]", prop_name, &index.to_string());
+                        parts.push_front(name.to_owned());
+                    }
+                    Some(PropKind::Map) => {
+                        match {
+                            let edge_weight = ctx
+                                .workspace_snapshot()?
+                                .edges_directed_for_edge_weight_kind(
+                                    attribute_value_id,
+                                    Direction::Incoming,
+                                    EdgeWeightKindDiscriminants::Contain,
+                                )
+                                .await?
+                                .pop()
+                                .ok_or(AttributeValueError::ParentAttributeValueMissing(
+                                    attribute_value_id,
+                                ))?;
+                            if let EdgeWeightKind::Contain(contain_key) = edge_weight.0.kind() {
+                                contain_key.to_owned()
+                            } else {
+                                None
+                            }
+                        } {
+                            Some(key) => {
+                                let name = format!("{0}[{1}]", prop_name, &key.to_owned());
+                                //let name = prop_name + &key.to_owned();
+                                parts.push_front(name.to_owned());
+                            }
+                            None => {
+                                parts.push_front(prop_name.to_owned());
+                            }
+                        }
+                    }
+                    Some(_) => parts.push_front(prop_name.to_owned()),
+                    None => continue,
+                }
+                attribute_value_id = parent_attribute_value_id;
+                work_queue.push_back(attribute_value_id);
+            } else {
+                // no more parents, need to add the prop_name to the parts because now we're at the root!
+                parts.push_front(prop_name);
+            }
+        }
+        if !parts.is_empty() {
+            Ok(Some(PropPath::new(parts).with_replaced_sep("/")))
+        } else {
+            Ok(None)
+        }
```

Comment:
I'd explain what's going on here in a comment and leave an additional space. I think what I've figured out is that we are assembling the parts here. I'm curious, why not use `PropPath` itself during the iterations rather than string operations?